### DISCUSSION TITLE
Snapshot testing with insta

### DISCUSSION THREAD

**Discussion by lcnbr** [ipetkov/crane#623] [Q&A] [answered]
I am trying to integrate crane into a rust crate that uses snapshot testing but I cannot figure out how to make it work. I am using insta for the snapshot tests, and when running the exact same cargo nextest command in my devshell I get no failures. However when running nix flake check, all snapshot tests fail. Does anyone have experience with this sort of setup?

See https://github.com/alphal00p/spenso for the flake.

**Comment by a-kenji** [comment, marked as answer]
Does your derivation have access to the snapshot test files?
It could be that:
```
 src = craneLib.cleanCargoSource (craneLib.path ./.);
```
filters them out and you need to include them through a [custom filter](https://crane.dev/source-filtering.html).

You could check that with `cntr`: https://github.com/Mic92/cntr and enter nix build sandbox.

**Reply by ipetkov** [reply]
I believe this is correct, you will need a custom source filter (as linked above) to include the `*.snap` files. By default `craneLib.cleanCargoSource` will only keep `*.rs` and `*.toml` files

**Reply by lcnbr** [reply]
I tried : 

```nix
src = lib.cleanSourceWith {
        src = craneLib.path ./.; # The original, unfiltered source
        filter = path: type: (craneLib.filterCargoSources path type) || (builtins.match ".*snap$" path != null);
      };
```

but that did not fix it..

**Reply by lcnbr** [reply]
Nevermind! It worked!
